cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -1317,6 +1317,45 @@ var waitFile = func(path string, wait time.Duration, n int) error {
 	return fmt.Errorf("no %v after waiting for %v", path, time.Duration(n)*wait)
 }
 
+// TODO: those have to be waited by udev instead
+func waitForDevice(path string) error {
+	if !osutil.FileExists(filepath.Join(dirs.GlobalRootDir, path)) {
+		pollWait := 50 * time.Millisecond
+		pollIterations := 1200
+		logger.Noticef("waiting up to %v for %v to appear", time.Duration(pollIterations)*pollWait, path)
+		if err := waitFile(filepath.Join(dirs.GlobalRootDir, path), pollWait, pollIterations); err != nil {
+			return fmt.Errorf("cannot find device: %v", err)
+		}
+	}
+	return nil
+}
+
+func getNonUEFISystemDisk(fallbacklabel string) (string, error) {
+	values, err := osutil.KernelCommandLineKeyValues("snapd_system_disk")
+	if err != nil {
+		return "", err
+	}
+	if value, ok := values["snapd_system_disk"]; ok {
+		if err := waitForDevice(value); err != nil {
+			return "", err
+		}
+		systemdDisk, err := disks.DiskFromDeviceName(value)
+		if err != nil {
+			systemdDiskDevicePath, errDevicePath := disks.DiskFromDevicePath(value)
+			if errDevicePath != nil {
+				return "", err
```

Comment:
This might be slightly silly, I wonder if we should return "err" or "errDevicePath" here